compiler/backend/src/org/jetbrains/kotlin/codegen/inline/SMAP.kt

```diff
@@ -171,6 +171,8 @@ object IdenticalSourceMapper : SourceMapper {
         get() = null
 
     override fun mapLineNumber(lineNumber: Int) = lineNumber
+
+    override fun mapLineNumber(source: Int, sourceName: String, sourcePath: String) = source
```

Comment:
This change appears unused... ?

Reply:
As I understand, this method should only be called if you have line numbers out of range for the current file. Thus, it's not correct to call it on IdenticalSourceMapper, and if that happens, it means you're missing some SMAPs. (See #2494, for example.)

Reply:
Thanks for the pointer! And your guess in the other comment was right; I saw some exceptions since a link to the parent function was missed in `FakeClassCodegen`; (re)storing it resolved the issue.

So, to prevent others from accidentally overriding it again, I wonder whether throwing an exception (yeah, super type already throws an exception but) with more detailed explanation makes sense.